# 국내 여행지역 관심도 정제

[국내 여행지역 관심도 데이터 요약]

- 링크: https://www.bigdata-culture.kr/bigdata/user/data_market/detail.do?id=359f0000-4129-11ec-a107-3b8bd6a15b10

- 출처: ㈜컨슈머인사이트, ‘주례(Weekly) 여행 행태 및 계획조사’(연간 26,000명)

- 조사 방식: 온라인(PC·모바일) 소비자 서베이

- 조사 내용:
    - 문항: “요즘 OOO을 가보고 싶다는 생각이 예전에 비해…” → 5점 척도로 응답
    - 응답자 특성 포함: 성별, 연령대, 거주지역, 소득수준 등
    - 조사 지역 (총 11개):
    서울, 경기인천, 부산, 충남대전, 충북, 경남울산, 경북대구, 전남광주, 전북, 강원, 제주

- 보유 데이터 기간: 2025년 1월 6일 ~ 6월 30일 (매주 단위 수집)

# 데이터 병합하기

In [176]:
import pandas as pd
import os
import glob

# 파일 경로 패턴 설정 (현재 작업 디렉토리에 있을 경우)
file_pattern = 'CI_DMSTC_TOUR_AREA_INTRST_DGREE_INFO_2025*.csv'

# 해당 패턴에 맞는 모든 파일 경로 리스트 (파일명이 오름차순으로 정렬)
file_list = sorted(glob.glob(file_pattern))

# 파일별 데이터프레임을 읽고 리스트로 저장
df_list = [pd.read_csv(file) for file in file_list]

# 모든 주차 데이터를 하나의 데이터프레임으로 병합
df_interest_2025 = pd.concat(df_list, ignore_index=True)

In [177]:
df_interest_2025.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6513 entries, 0 to 6512
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   RESPOND_ID                6513 non-null   int64 
 1   EXAMIN_BEGIN_DE           6513 non-null   int64 
 2   SEXDSTN_FLAG_CD           6513 non-null   object
 3   AGRDE_FLAG_NM             6513 non-null   object
 4   ANSWRR_OC_AREA_NM         6513 non-null   object
 5   HSHLD_INCOME_DGREE_NM     6513 non-null   object
 6   SU_TOUR_INTRST_VALUE      6513 non-null   object
 7   GG_IC_TOUR_INTRST_VALUE   6513 non-null   object
 8   BS_TOUR_INTRST_VALUE      6513 non-null   object
 9   CHN_DJ_TOUR_INTRST_VALUE  6513 non-null   object
 10  CB_TOUR_INTRST_VALUE      6513 non-null   object
 11  GN_US_TOUR_INTRST_VALUE   6513 non-null   object
 12  GB_DG_TOUR_INTRST_VALUE   6513 non-null   object
 13  JN_GJ_TOUR_INTRST_VALUE   6513 non-null   object
 14  JNBK_TOUR_INTRST_VALUE  

In [178]:
df_interest_2025

RESPOND_ID  EXAMIN_BEGIN_DE SEXDSTN_FLAG_CD AGRDE_FLAG_NM  \
0       53329624         20250106               F           60대   
1       53336932         20250106               F           60대   
2       53329552         20250106               F           50대   
3       53365353         20250106               F           50대   
4       53358562         20250106               F           50대   
...          ...              ...             ...           ...   
6508    53375520         20250630               F           20대   
6509    53375590         20250630               F           20대   
6510    53375667         20250630               F           60대   
6511    53376802         20250630               F           50대   
6512    53377011         20250630               M           40대   

     ANSWRR_OC_AREA_NM HSHLD_INCOME_DGREE_NM SU_TOUR_INTRST_VALUE  \
0                 충청북도              700만원 이상               약간 커졌다   
1                 충청북도         300이상500만원 미만               약간 커졌다   
2                부산광역시         500이상700만원 미만             예전과 비슷하다   
3                부산광역시         300이상500만원 미만              약간 적어졌다   
4                인천광역시              300만원 미만             예전과 비슷하다   
...                ...                   ...                  ...   
6508      충청남도(세종시 포함)                    모름              많이 적어졌다   
6509               경기도                    모름             예전과 비슷하다   
6510               경기도         500이상700만원 미만             예전과 비슷하다   
6511               경기도              700만원 이상             예전과 비슷하다   
6512              충청북도              700만원 이상             예전과 비슷하다   

     GG_IC_TOUR_INTRST_VALUE BS_TOUR_INTRST_VALUE CHN_DJ_TOUR_INTRST_VALUE  \
0                   예전과 비슷하다               약간 커졌다                 예전과 비슷하다   
1                     약간 커졌다               약간 커졌다                   약간 커졌다   
2                     약간 커졌다               약간 커졌다                 예전과 비슷하다   
3                   예전과 비슷하다              약간 적어졌다                 예전과 비슷하다   
4                    약간 적어졌다             예전과 비슷하다                  약간 적어졌다   
...                      ...                  ...                      ...   
6508                 많이 적어졌다              많이 적어졌다                   약간 커졌다   
6509                예전과 비슷하다             예전과 비슷하다                 예전과 비슷하다   
6510                  약간 커졌다              약간 적어졌다                 예전과 비슷하다   
6511                  많이 커졌다               약간 커졌다                 예전과 비슷하다   
6512                예전과 비슷하다             예전과 비슷하다                   약간 커졌다   

     CB_TOUR_INTRST_VALUE GN_US_TOUR_INTRST_VALUE GB_DG_TOUR_INTRST_VALUE  \
0                예전과 비슷하다                예전과 비슷하다                예전과 비슷하다   
1                예전과 비슷하다                  약간 커졌다                  약간 커졌다   
2                예전과 비슷하다                  약간 커졌다                  약간 커졌다   
3                예전과 비슷하다                 약간 적어졌다                예전과 비슷하다   
4                예전과 비슷하다                 약간 적어졌다                예전과 비슷하다   
...                   ...                     ...                     ...   
6508              많이 적어졌다                 많이 적어졌다                 많이 적어졌다   
6509              약간 적어졌다                예전과 비슷하다                예전과 비슷하다   
6510               약간 커졌다                  약간 커졌다                  약간 커졌다   
6511               약간 커졌다                  약간 커졌다                  약간 커졌다   
6512             예전과 비슷하다                예전과 비슷하다                예전과 비슷하다   

     JN_GJ_TOUR_INTRST_VALUE JNBK_TOUR_INTRST_VALUE GW_TOUR_INTRST_VALUE  \
0                     약간 커졌다               예전과 비슷하다               약간 커졌다   
1                     약간 커졌다               예전과 비슷하다               약간 커졌다   
2                   예전과 비슷하다               예전과 비슷하다             예전과 비슷하다   
3                   예전과 비슷하다                 약간 커졌다               약간 커졌다   
4                   예전과 비슷하다                약간 적어졌다               많이 커졌다   
...                      ...                    ...   

In [ ]:
# 병합된 데이터 저장 (인코딩은 utf-8-sig로 한글 깨짐 방지)
df_interest_2025.to_csv('국내여행지역관심도_2025상반기_0729.csv', index=False, encoding='utf-8-sig')

In [180]:
# 중복된 RESPOND_ID 수 확인
duplicate_count = df_interest_2025['RESPOND_ID'].duplicated().sum()
print(f"🔍 중복된 RESPOND_ID 개수: {duplicate_count}")

🔍 중복된 RESPOND_ID 개수: 904


- 동일한 사람이 여러 번 응답했음을 의미

In [181]:
# 전체 행 기준 완전 중복 확인
duplicate_rows = df_interest_2025.duplicated()
print(f"✅ 완전 동일한 행 개수: {duplicate_rows.sum()}")


✅ 완전 동일한 행 개수: 0


- ID가 같더라도 응답 내용은 모두 다름 → 중복 응답 아님
- 동일인이 여러 주차에 걸쳐 서로 다른 응답을 한 것으로 간주할 수 있으므로 데이터 그대로 사용

# 변수 정리

1. RESPOND_ID               : 응답자 ID  
2. EXAMIN_BEGIN_DE          : 조사시작일자  
3. SEXDSTN_FLAG_CD          : 성별구분코드  
4. AGRDE_FLAG_NM            : 연령대구분명  
5. ANSWRR_OC_AREA_NM        : 답변자 거주지역명  
6. HSHLD_INCOME_DGREE_NM    : 가구소득정도명  
7. SU_TOUR_INTRST_VALUE     : 서울특별시 여행관심값  
8. GG_IC_TOUR_INTRST_VALUE  : 경기도/인천광역시 여행관심값  
9. BS_TOUR_INTRST_VALUE     : 부산광역시 여행관심값  
10. CHN_DJ_TOUR_INTRST_VALUE: 충청남도/대전광역시 여행관심값  
11. CB_TOUR_INTRST_VALUE    : 충청북도 여행관심값  
12. GN_US_TOUR_INTRST_VALUE : 경상남도/울산광역시 여행관심값  
13. GB_DG_TOUR_INTRST_VALUE : 경상북도/대구광역시 여행관심값  
14. JN_GJ_TOUR_INTRST_VALUE : 전라남도/광주광역시 여행관심값  
15. JNBK_TOUR_INTRST_VALUE  : 전라북도 여행관심값  
16. GW_TOUR_INTRST_VALUE    : 강원도 여행관심값  
17. JJ_TOUR_INTRST_VALUE    : 제주도 여행관심값

# 데이터 정제 시작

## 지역별 관심도 값만 남기고 나머지 열 제거

In [182]:
# 지역별 여행 관심도 컬럼만 추출
region_cols = [
    'SU_TOUR_INTRST_VALUE',     # 서울
    'GG_IC_TOUR_INTRST_VALUE',  # 경기인천
    'BS_TOUR_INTRST_VALUE',     # 부산
    'CHN_DJ_TOUR_INTRST_VALUE', # 충남대전
    'CB_TOUR_INTRST_VALUE',     # 충북
    'GN_US_TOUR_INTRST_VALUE',  # 경남울산
    'GB_DG_TOUR_INTRST_VALUE',  # 경북대구
    'JN_GJ_TOUR_INTRST_VALUE',  # 전남광주
    'JNBK_TOUR_INTRST_VALUE',   # 전북
    'GW_TOUR_INTRST_VALUE',     # 강원
    'JJ_TOUR_INTRST_VALUE'      # 제주
]

# 정제된 데이터프레임 이름: df
df = df_interest_2025[region_cols]

In [183]:
df

SU_TOUR_INTRST_VALUE GG_IC_TOUR_INTRST_VALUE BS_TOUR_INTRST_VALUE  \
0                  약간 커졌다                예전과 비슷하다               약간 커졌다   
1                  약간 커졌다                  약간 커졌다               약간 커졌다   
2                예전과 비슷하다                  약간 커졌다               약간 커졌다   
3                 약간 적어졌다                예전과 비슷하다              약간 적어졌다   
4                예전과 비슷하다                 약간 적어졌다             예전과 비슷하다   
...                   ...                     ...                  ...   
6508              많이 적어졌다                 많이 적어졌다              많이 적어졌다   
6509             예전과 비슷하다                예전과 비슷하다             예전과 비슷하다   
6510             예전과 비슷하다                  약간 커졌다              약간 적어졌다   
6511             예전과 비슷하다                  많이 커졌다               약간 커졌다   
6512             예전과 비슷하다                예전과 비슷하다             예전과 비슷하다   

     CHN_DJ_TOUR_INTRST_VALUE CB_TOUR_INTRST_VALUE GN_US_TOUR_INTRST_VALUE  \
0                    예전과 비슷하다             예전과 비슷하다                예전과 비슷하다   
1                      약간 커졌다             예전과 비슷하다                  약간 커졌다   
2                    예전과 비슷하다             예전과 비슷하다                  약간 커졌다   
3                    예전과 비슷하다             예전과 비슷하다                 약간 적어졌다   
4                     약간 적어졌다             예전과 비슷하다                 약간 적어졌다   
...                       ...                  ...                     ...   
6508                   약간 커졌다              많이 적어졌다                 많이 적어졌다   
6509                 예전과 비슷하다              약간 적어졌다                예전과 비슷하다   
6510                 예전과 비슷하다               약간 커졌다                  약간 커졌다   
6511                 예전과 비슷하다               약간 커졌다                  약간 커졌다   
6512                   약간 커졌다             예전과 비슷하다                예전과 비슷하다   

     GB_DG_TOUR_INTRST_VALUE JN_GJ_TOUR_INTRST_VALUE JNBK_TOUR_INTRST_VALUE  \
0                   예전과 비슷하다                  약간 커졌다               예전과 비슷하다   
1                     약간 커졌다                  약간 커졌다               예전과 비슷하다   
2                     약간 커졌다                예전과 비슷하다               예전과 비슷하다   
3                   예전과 비슷하다                예전과 비슷하다                 약간 커졌다   
4                   예전과 비슷하다                예전과 비슷하다                약간 적어졌다   
...                      ...                     ...                    ...   
6508                 많이 적어졌다                  약간 커졌다                약간 적어졌다   
6509                예전과 비슷하다                예전과 비슷하다               예전과 비슷하다   
6510                  약간 커졌다                  약간 커졌다               예전과 비슷하다   
6511                  약간 커졌다                  약간 커졌다                 약간 커졌다   
6512                예전과 비슷하다                예전과 비슷하다               예전과 비슷하다   

     GW_TOUR_INTRST_VALUE JJ_TOUR_INTRST_VALUE  
0                  약간 커졌다               많이 커졌다  
1                  약간 커졌다               약간 커졌다  
2                예전과 비슷하다             예전과 비슷하다  
3                  약간 커졌다             예전과 비슷하다  
4                  많이 커졌다               많이 커졌다  
...                   ...                  ...  
6508               약간 커졌다              많이 적어졌다  
6509               약간 커졌다              약간 적어졌다  
6510             예전과 비슷하다               약간 커졌다  
6511               약간 커졌다               약간 커졌다  
6512               약간 커졌다             예전과 비슷하다  

[6513 rows x 11 columns]

In [184]:
# 컬럼명을 한글 지역명으로 변경
df.columns = [
    '서울',
    '경기인천',
    '부산',
    '충남대전',
    '충북',
    '경남울산',
    '경북대구',
    '전남광주',
    '전북',
    '강원',
    '제주'
]

In [185]:
df

서울      경기인천        부산      충남대전        충북      경남울산      경북대구  \
0       약간 커졌다  예전과 비슷하다    약간 커졌다  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다   
1       약간 커졌다    약간 커졌다    약간 커졌다    약간 커졌다  예전과 비슷하다    약간 커졌다    약간 커졌다   
2     예전과 비슷하다    약간 커졌다    약간 커졌다  예전과 비슷하다  예전과 비슷하다    약간 커졌다    약간 커졌다   
3      약간 적어졌다  예전과 비슷하다   약간 적어졌다  예전과 비슷하다  예전과 비슷하다   약간 적어졌다  예전과 비슷하다   
4     예전과 비슷하다   약간 적어졌다  예전과 비슷하다   약간 적어졌다  예전과 비슷하다   약간 적어졌다  예전과 비슷하다   
...        ...       ...       ...       ...       ...       ...       ...   
6508   많이 적어졌다   많이 적어졌다   많이 적어졌다    약간 커졌다   많이 적어졌다   많이 적어졌다   많이 적어졌다   
6509  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다   약간 적어졌다  예전과 비슷하다  예전과 비슷하다   
6510  예전과 비슷하다    약간 커졌다   약간 적어졌다  예전과 비슷하다    약간 커졌다    약간 커졌다    약간 커졌다   
6511  예전과 비슷하다    많이 커졌다    약간 커졌다  예전과 비슷하다    약간 커졌다    약간 커졌다    약간 커졌다   
6512  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다    약간 커졌다  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다   

          전남광주        전북        강원        제주  
0       약간 커졌다  예전과 비슷하다    약간 커졌다    많이 커졌다  
1       약간 커졌다  예전과 비슷하다    약간 커졌다    약간 커졌다  
2     예전과 비슷하다  예전과 비슷하다  예전과 비슷하다  예전과 비슷하다  
3     예전과 비슷하다    약간 커졌다    약간 커졌다  예전과 비슷하다  
4     예전과 비슷하다   약간 적어졌다    많이 커졌다    많이 커졌다  
...        ...       ...       ...       ...  
6508    약간 커졌다   약간 적어졌다    약간 커졌다   많이 적어졌다  
6509  예전과 비슷하다  예전과 비슷하다    약간 커졌다   약간 적어졌다  
6510    약간 커졌다  예전과 비슷하다  예전과 비슷하다    약간 커졌다  
6511    약간 커졌다    약간 커졌다    약간 커졌다    약간 커졌다  
6512  예전과 비슷하다  예전과 비슷하다    약간 커졌다  예전과 비슷하다  

[6513 rows x 11 columns]

## 답변을 5점 척도로 변경

In [186]:
# 전체 df에서 사용된 답변 유형 확인
unique_values = pd.unique(df.values.ravel())
print(unique_values)
print(f"총 응답 유형 개수: {len(unique_values)}")

['약간 커졌다' '예전과 비슷하다' '많이 커졌다' '약간 적어졌다' '많이 적어졌다']
총 응답 유형 개수: 5


- '많이 커졌다' -> 5
- '약간 커졌다' -> 4
- '예전과 비슷하다' -> 3
- '약간 적어졌다' -> 2
- '많이 적어졌다' -> 1

In [187]:
# 문자 응답을 5점 척도로 매핑
scale_map = {
    '많이 커졌다': 5,
    '약간 커졌다': 4,
    '예전과 비슷하다': 3,
    '약간 적어졌다': 2,
    '많이 적어졌다': 1
}

# df 전체에 적용 (한 번에)
df = df.replace(scale_map)

C:\Users\user\AppData\Local\Temp\ipykernel_13444\2890665579.py:11: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df = df.replace(scale_map)


In [188]:
df

서울  경기인천  부산  충남대전  충북  경남울산  경북대구  전남광주  전북  강원  제주
0      4     3   4     3   3     3     3     4   3   4   5
1      4     4   4     4   3     4     4     4   3   4   4
2      3     4   4     3   3     4     4     3   3   3   3
3      2     3   2     3   3     2     3     3   4   4   3
4      3     2   3     2   3     2     3     3   2   5   5
...   ..   ...  ..   ...  ..   ...   ...   ...  ..  ..  ..
6508   1     1   1     4   1     1     1     4   2   4   1
6509   3     3   3     3   2     3     3     3   3   4   2
6510   3     4   2     3   4     4     4     4   3   3   4
6511   3     5   4     3   4     4     4     4   4   4   4
6512   3     3   3     4   3     3     3     3   3   4   3

[6513 rows x 11 columns]

In [189]:
# 1. 각 지역별 점수별 응답자 수 집계
score_counts = df.apply(pd.Series.value_counts).fillna(0).astype(int)

# 2. 지역이 행, 점수가 열인 형태로 보기 쉽게 정리
score_counts = score_counts.T  # 전치: 지역이 index, 점수가 column

# 3. 결과 확인
print(score_counts)


        1     2     3     4    5
서울    486   707  3669  1250  401
경기인천  503   906  3812  1135  157
부산    463   731  3223  1741  355
충남대전  519   926  3770  1163  135
충북    595  1050  4002   781   85
경남울산  722  1164  3470  1043  114
경북대구  944  1155  3420   895   99
전남광주  681   962  3268  1377  225
전북    674  1055  3599  1049  136
강원    374   579  2906  2172  482
제주    729   970  2534  1727  553


## 지역별 평균 관심도 점수 계산

In [190]:
# ✅ 각 지역별 평균 관심도 점수 계산
# - df는 지역별 1~5점 관심도 데이터프레임
# - 각 지역(컬럼)별로 전체 응답자의 평균 점수를 계산
region_mean = df.mean()

# ✅ 평균 점수를 요약 데이터프레임 df1으로 변환
# - 컬럼명: 지역, 지역별_관심도
df1 = region_mean.reset_index()
df1.columns = ['지역', '지역별_관심도']


In [191]:
df1

지역   지역별_관심도
0     서울  3.057270
1   경기인천  2.928911
2     부산  3.121910
3   충남대전  2.918471
4     충북  2.802088
5   경남울산  2.794718
6   경북대구  2.700599
7   전남광주  2.923691
8     전북  2.833871
9     강원  3.277752
10    제주  3.062183

## 변수명 변경

In [192]:
# 변수명 변경
df1.columns = ['지역', '지역별_관심도']

In [193]:
df1

지역   지역별_관심도
0     서울  3.057270
1   경기인천  2.928911
2     부산  3.121910
3   충남대전  2.918471
4     충북  2.802088
5   경남울산  2.794718
6   경북대구  2.700599
7   전남광주  2.923691
8     전북  2.833871
9     강원  3.277752
10    제주  3.062183

## 이중 지역인 것 분리

In [194]:
# 복사본 생성
df2 = df1.copy()

# append 대신 concat을 위한 리스트 준비
rows_to_add = []

# 묶인 지역 → 분리 지역 딕셔너리
split_map = {
    '경기인천': ['경기', '인천'],
    '충남대전': ['충남', '대전'],
    '경남울산': ['경남', '울산'],
    '경북대구': ['경북', '대구'],
    '전남광주': ['전남', '광주']
}

# 각 묶인 지역 값 → 분리 지역으로 복사
for combined, splitted in split_map.items():
    value = df1.loc[df1['지역'] == combined, '지역별_관심도'].values[0]
    for region in splitted:
        rows_to_add.append({'지역': region, '지역별_관심도': value})

# 새 지역 행 추가
df2 = pd.concat([df1, pd.DataFrame(rows_to_add)], ignore_index=True)

# 기존 묶인 지역 제거 (← 여기 수정됨!)
df2 = df2[~df2['지역'].isin(split_map.keys())].reset_index(drop=True)


In [195]:
df2

지역   지역별_관심도
0   서울  3.057270
1   부산  3.121910
2   충북  2.802088
3   전북  2.833871
4   강원  3.277752
5   제주  3.062183
6   경기  2.928911
7   인천  2.928911
8   충남  2.918471
9   대전  2.918471
10  경남  2.794718
11  울산  2.794718
12  경북  2.700599
13  대구  2.700599
14  전남  2.923691
15  광주  2.923691

- 세종시는 통계 누락으로 인해 지리적 및 생활권 유사성이 큰 **대전시의 관심도 값(2.918471)**을 사용함

In [196]:
# 1. 대전의 관심도 값을 가져옴
sejong_value = df2[df2['지역'] == '대전']['지역별_관심도'].values[0]

# 2. 세종 행을 df2에 추가함
df2.loc[len(df2)] = ['세종', sejong_value]

In [197]:
df2

지역   지역별_관심도
0   서울  3.057270
1   부산  3.121910
2   충북  2.802088
3   전북  2.833871
4   강원  3.277752
5   제주  3.062183
6   경기  2.928911
7   인천  2.928911
8   충남  2.918471
9   대전  2.918471
10  경남  2.794718
11  울산  2.794718
12  경북  2.700599
13  대구  2.700599
14  전남  2.923691
15  광주  2.923691
16  세종  2.918471

In [198]:
# df2 저장 (CSV 형식, 인코딩 UTF-8)
df2.to_csv('지역별_관심도_최종.csv', index=False, encoding='utf-8-sig')

In [ ]:
import pandas as pd
# df3 불러오기
df3 = pd.read_csv('지역별_관심도_최종_0729.csv')
df3

지역   지역별_관심도
0   서울  3.057270
1   부산  3.121910
2   충북  2.802088
3   전북  2.833871
4   강원  3.277752
5   제주  3.062183
6   경기  2.928911
7   인천  2.928911
8   충남  2.918471
9   대전  2.918471
10  경남  2.794718
11  울산  2.794718
12  경북  2.700599
13  대구  2.700599
14  전남  2.923691
15  광주  2.923691
16  세종  2.918471